indico/modules/legal/__init__.py

```diff
@@ -51,3 +56,34 @@ def _inject_privacy_footer(**kwargs):
     url = legal_settings.get('privacy_policy_url')
     if url or legal_settings.get('privacy_policy'):
         return render_template('legal/privacy_footer.html', url=url)
+
+
+@signals.core.app_created.connect
+def _handle_terms_changed(app, **kwargs):
+    @app.before_request
+    def _terms_if_not_accepted():
+        terms_date = legal_settings.get('terms_effective_date')
+        if not legal_settings.get('terms_require_accept') or not terms_date:
+            return
+
+        # If you are not logged in or have accepted the most current terms, you are good
+        if not session.user:
+            return
+
+        if session.user.accepted_tos_dt and session.user.accepted_tos_dt >= terms_date:
+            return
+
+        if request.endpoint:
+            prefix = request.endpoint.split('.', 1)[0]
+
+            # Certain endpoints need to be passed through.
+            if prefix in ('assets', 'legal', 'auth', 'core'):
+                return
+
+            # Admins get a free pass on the admin interface
+            if session.user.is_admin and request.path.startswith('/admin/'):
+                return
```

Comment:
TBH I'd let them completely ignore this and just show a persistent notification banner on top

Reply:
For banners on top? The global announcement or user impersonation status for example.